File: Algorithms/Searches/python/linearsearch.py
import os.path
import random

def linear_search(nums, target):
  print(f"Searching for target: {target}.")
  count = 0
  for number in nums:
    if number == target:
      print(f"Target found in {count + 1} tries.")
      return
    count += 1
  print("Target was not found.")

def main():
  current_path = os.path.abspath(os.path.dirname(__file__))
  path = os.path.join(current_path, "../../Tests/sortednumbers.txt")
  with open(path) as file:
    nums = []
    for number in file.readlines():
      nums.append(int(number))
  file.close()

  for i in range(10):
    target = random.randrange(-20000, 20000)
    linear_search(nums, target)

main()

File: Algorithms/Searches/cpp/linearsearch.cpp
#include <iostream>
#include <fstream>
#include <time.h>
#include <array>

using namespace std;

int main()
{
  ifstream nums("../../Tests/sortednumbers.txt");
  const int size = 10000;

  int c = 0;
  int num;
  int *array = new int[size];

  while (nums >> num && c < size)
  {
    array[c++] = num;
  }

  for (int i; i < 100; i++)
  {
    cout << array[i] << endl;
  }

  nums.close();

  srand(time(0));

  int min = -20000;
  int max = 20000;

  num = (rand() % (max - min + 1)) + min;

  cout << "Searching for number '" << num << "'." << endl;

  bool found = false;
  int count;

  for (int i; i < size; i++)
  {
    if (array[i] == num)
    {
      found = true;
      count = i + 1;
      break;
    }

    continue;
  }

  if (found)
  {
    cout << "Number found in " << count << " tries.";
  }
  else
  {
    cout << "Number not found.";
  }

  return 0;
}